### PR TITLE
GBS: Polish vibronic docstrings

### PR DESCRIPTION
**Pull request by ixfoduap**
Makes changes to docstrings for all code related to vibronic spectra. In particular, it significantly shortens docstrings for the vibronic module to coincide with the rest of the documentation.

### CODE REVIEW

File: strawberryfields/gbs/vibronic.py [hunk 1]
```diff
@@ -172,7 +99,7 @@ def gbs_params(
         w (array): normal mode frequencies of the initial electronic state (:math:`\mbox{cm}^{-1}`)
         wp (array): normal mode frequencies of the final electronic state (:math:`\mbox{cm}^{-1}`)
         Ud (array): Duschinsky matrix
-        d (array): Duschinsky displacement vector corrected with wp
+        d (array): Duschinsky displacement vector
```
Comment (soranjh) [inline strawberryfields/gbs/vibronic.py]: The term d here is actually delta which is related to Duschinsky displacement d as:
delta = sqrt(w / h-bar) * d, with w being the angular frequency. We should probably explain it somewhere.

File: strawberryfields/gbs/vibronic.py [hunk 2]
```diff
@@ -19,120 +19,47 @@
 
 .. currentmodule:: strawberryfields.gbs.vibronic
 
-This module contains functions for computing molecular vibronic spectra using GBS.
+This module contains functions used for computing molecular vibronic spectra with GBS. In vibronic
+spectroscopy, incoming light simultaneously excites a molecule to higher electronic and
+vibrational states. The difference in the energies of the ground and excited states
```
Comment (soranjh) [inline strawberryfields/gbs/vibronic.py]: We might say initial and final states to be safe.

Comment (ixfoduap) [inline strawberryfields/gbs/vibronic.py]:
```suggestion
vibrational states. The difference in the energies of the initial and final states
```

File: strawberryfields/gbs/vibronic.py [hunk 3]
```diff
@@ -19,120 +19,47 @@
...
-This module contains functions for computing molecular vibronic spectra using GBS.
+This module contains functions used for computing molecular vibronic spectra with GBS. In vibronic
+spectroscopy, incoming light simultaneously excites a molecule to higher electronic and
+vibrational states. The difference in the energies of the initial and final states
+determines the frequency of the photons that are absorbed by the molecule. The probability of
+the transition between two specific vibrational states are referred to as Franck-Condon factors
+(FCFs). They determine the intensity of absorption peaks in a vibronic spectrum. A GBS
+device can be programmed to provide samples that can be processed to obtain the molecular
+vibronic spectra. Theoretical background on computing vibronic spectra using GBS can be found in
+:cite:`huh2015boson` and :cite:`quesada2019franck`.
```
Comment (soranjh) [inline strawberryfields/gbs/vibronic.py]:
```suggestion
:cite:`Huh_2015` and :cite:`quesada2019franck`.
```

**Pull request merged**